hathor/p2p/manager.py

```diff
@@ -223,13 +229,24 @@ def on_peer_disconnect(self, protocol: HathorProtocol) -> None:
                 self.connected_peers[protocol.peer.id] = existing_protocol
         self.pubsub.publish(HathorEvents.NETWORK_PEER_DISCONNECTED, protocol=protocol)
 
+    # XXX: maybe rename these to `iter_*` since they aren't returning a contatainer
+
     def get_all_connections(self) -> Iterable[HathorProtocol]:
         """Iterate over all connections."""
         for conn in self.connections:
             yield conn
 
-    def get_ready_connections(self) -> Set[HathorProtocol]:
-        return set(self.connected_peers.values())
+    def get_ready_connections(self) -> Iterable[HathorProtocol]:
+        """Iterate over ready connections."""
+        for conn in self.connected_peers.values():
+            yield conn
+
+    def get_not_ready_connections(self) -> Iterable[HathorProtocol]:
+        """Iterate over not-ready connections."""
+        for conn in self.connecting_peers.values():
```

Comment:
Yes. I think `connecting_peers` could have the endpoint url string (in addition to, or instead of the `IStreamClientEndpoint`), and `get_not_ready_connections` could yield `str`s of `endpoint_url`s or `connection_string`s. What do you think?

Reply:
Fixed in #292 .